jdrfandroidbleparser/src/test/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementContextCharacteristicTest.kt

```diff
@@ -1,13 +1,200 @@
+    fun testParseSuccess() {
+        Assert.assertTrue(glucoseMeasurementContext1.successfulParsing)
+        Assert.assertTrue(glucoseMeasurementContext2.successfulParsing)
+        Assert.assertTrue(glucoseMeasurementContext3.successfulParsing)
+        Assert.assertTrue(glucoseMeasurementContext4.successfulParsing)
+        Assert.assertTrue(glucoseMeasurementContext5.successfulParsing)
+    }
+
+    @Test
+    fun testParseSequenceNumber() {
+        Assert.assertEquals(sequenceNumber1, glucoseMeasurementContext1.sequenceNumber)
+        Assert.assertEquals(sequenceNumber2, glucoseMeasurementContext2.sequenceNumber)
+        Assert.assertEquals(sequenceNumber3, glucoseMeasurementContext3.sequenceNumber)
+        Assert.assertEquals(sequenceNumber4, glucoseMeasurementContext4.sequenceNumber)
+        Assert.assertEquals(sequenceNumber5, glucoseMeasurementContext5.sequenceNumber)
+    }
+
+    @Test
+    fun testParseExtendedFlag() {
+        Assert.assertEquals(extendedFlag1, glucoseMeasurementContext1.extendedFlag)
+        Assert.assertEquals(extendedFlag3, glucoseMeasurementContext3.extendedFlag)
+        Assert.assertEquals(extendedFlag4, glucoseMeasurementContext4.extendedFlag)
+        Assert.assertEquals(extendedFlag5, glucoseMeasurementContext5.extendedFlag)
+    }
+
+    @Test
+    fun testParseCarbohydrates() {
+        Assert.assertEquals(carbType2, glucoseMeasurementContext2.carbohydrateId)
+        Assert.assertEquals(carbAmount2, glucoseMeasurementContext2.carbohydrate)
+        Assert.assertEquals(carbType5, glucoseMeasurementContext5.carbohydrateId)
+        Assert.assertEquals(carbAmount5, glucoseMeasurementContext5.carbohydrate)
+    }
+
+    @Test
+    fun testParseMeal() {
+        Assert.assertEquals(mealType2, glucoseMeasurementContext2.meal)
+        Assert.assertEquals(mealType3, glucoseMeasurementContext3.meal)
+        Assert.assertEquals(mealType5, glucoseMeasurementContext5.meal)
+    }
+
+    @Test
+    fun testParseTesterHealth() {
+        Assert.assertEquals(tester1, glucoseMeasurementContext1.tester)
+        Assert.assertEquals(tester2, glucoseMeasurementContext2.tester)
+        Assert.assertEquals(tester3, glucoseMeasurementContext3.tester)
+        Assert.assertEquals(tester4, glucoseMeasurementContext4.tester)
+        Assert.assertEquals(tester5, glucoseMeasurementContext5.tester)
+
+        Assert.assertEquals(health1, glucoseMeasurementContext1.health)
+        Assert.assertEquals(health2, glucoseMeasurementContext2.health)
+        Assert.assertEquals(health3, glucoseMeasurementContext3.health)
+        Assert.assertEquals(health4, glucoseMeasurementContext4.health)
+        Assert.assertEquals(health5, glucoseMeasurementContext5.health)
+    }
+
+    @Test
+    fun testParseExerciseDurationAndIntensity() {
+        Assert.assertEquals(exerciseDuration3, glucoseMeasurementContext3.exerciseDuration)
+        Assert.assertEquals(exerciseIntesity3, glucoseMeasurementContext3.exerciseIntensity)
```

Comment:
A remark, good to see it support float compares now.

Reply:
👍